examples/react-apollo/cypress/e2e/2-sign-up/anonymous.cy.ts

```diff
@@ -0,0 +1,54 @@
+import { faker } from '@faker-js/faker'
+
+context('Anonymous users', () => {
+  beforeEach(() => {
+    cy.signInAnonymous()
+  })
+
+  it('should sign-up anonymously', () => {
+    cy.contains('You are authenticated')
+    cy.fetchUserData().its('isAnonymous').should('equal', true)
+    cy.contains('You signed in anonymously')
+  })
+
+  it('should deanonymise with email+password', () => {
+    cy.fetchUserData()
```

Comment:
Isn't there a better way to get user information on the UI? This looks like a huge implementation detail that is being used here. We should use something here that would mirror how a user would get this information from the backend. (e.g: clicking some button or something like that)

Reply:
We should also not test the actual user object, but the UI instead whether it changed correctly thanks to the deanonymization.

Reply:
Okay, it makes sense then. Can we convert this to async / await btw? That would make this code more readable IMO

Reply:
What do you mean?
https://docs.cypress.io/faq/questions/using-cypress-faq#Can-I-use-the-new-ES7-async-await-syntax
https://github.com/cypress-io/eslint-plugin-cypress/blob/master/docs/rules/no-async-tests.md

Reply:
oh, sure